# Exercise 0: Google your favorite comic strip ever. Make a markdown cell that contains the following: 
* Top level heading that says "My favorite comic ever"
* Separate bullet points for the name of the comic (in bold text), the name of the author (in italic text), and the year it was published
* Insert a hyperlink to the comic
* Bonus: google "embed image jupyter notebook" and click on the first stackoverflow link (should be top link, for me at least). Figure out how to embed the image directly into your notebook

# My favorite comic ever
* [**Dilbert: Brain Golfing**](https://dilbert.com/search_results?terms=Brain+Golfing)
* *Scott Adams*
* 2010
![title](brain_golfing.gif)

## Exercise 1: Assign the following strings to variables and then print them out
* My name is Ted
* Roy is old, but still a better leader than Jamie
* In response, he said "but what if I wear the red trainers?"
* They both love Keeley, what'll she do???

Then take your last string (i.e., They both love Keeley, what'll she do???) and print the 1st letter (T), the fourth to last letter (o), and the 6th letter (b)


In [7]:
s1 = 'My name is Ted'
s2 = 'Roy is old,but still a better leader than Jamie'
s3 = 'In response, he said "but what if I wear the read trainers?"'
s4 = "They both love Keeley, but what'll she do???"
print(s1) 
print(s2)
print(s3)
print(s4)
print(s4[0])
print(s4[-4])
print(s4[5])

My name is Ted
Roy is old,but still a better leader than Jamie
In response, he said "but what if I wear the read trainers?"
They both love Keeley, but what'll she do???
T
o
b


## Exercise 2

### Print the number of *seconds* it would take you to fold 1000 pizza boxes if you were able to maintain the world record pace of 14 pizza boxes in one minute indefinitely.
* [link here](https://www.guinnessworldrecords.com/world-records/103587-most-pizza-boxes-folded-in-one-minute)

**Bonus**: see if you can find a way to print the decimal only to the tenth's place. Google "python round" and see if you can figure it out

In [8]:
boxes = 1000
rate = 14
minutes = boxes/rate
seconds = minutes *60
print(seconds)
round(seconds, 1)

4285.714285714286


4285.7

Correct output: 4285.714285714286 (4285.7 for bonus)

## Exercise 3

### Imagine you have a data set that is 'csv' format (comma separate value) where each data point (a word) is seperated by commas. Given the example string below, alphabetize the words and print the last word in the alphabetized list. Hint: first convert to a list (hint in a hint - use 'split'), then sort the list, the find the last entry. 

In [1]:
words = 'banana,peanut,xylophone,aardvark,foot,python'
# insert code here
lst = words.split(',')
lst.sort()
print(lst[-1])

xylophone


Correct output: xylophone

## Exercise 4 (Challenge Problem - data cleaning)

### Now imagine that you have a data set where the words are already stored in a list with each element being one word, but there is an inexplicable number of spaces at the start or end of each word. Before you can deal with the data you need to clean the spaces so your can determine the shortest word in the list.

Hints:
* Use 'join' to convert the list into a single string. 
* Remove all the blank spaces using the 'replace' method [here](https://www.geeksforgeeks.org/python-string-replace/). Note that " " is how you indicate an empty space, so replace all of those with just "" (nothing)
* Use 'split' to convert your single string back into a list (split based on ',')
* The *sorted* function has an optional argument *key* that you can pass a function to such that *sorted* uses that function to determine the order of sorting (see documentation [here](https://docs.python.org/3/howto/sorting.html)). Sort based on word length (len)

In [68]:
words_lst = ['banana ', ' peanut  ', 'xylophone  ', 'aardvark  ', 'foot       ', 'python    ']
# insert code here
words = ','.join(words_lst)
nospace = words.replace(' ','')
lst = nospace.split(',')
srt = sorted(lst, key=len)
print(srt[0])

foot


Correct output: foot
